ry5n/stencil-test
# Write Spec component

Builds out a full implementation of a Spec component.

Status: **Ready to merge**

Reviewers: TBD
Ticket: Part of [CSOPS-1275](https://mobify.atlassian.net/browse/CSOPS-1257)
Linked PRs: dustjs-helper-component/pull/2
## Changes
- Adds stylesheet and schema, removes unnecessary ui script
- Adds dust templates compatible with the experimental Dust component extensions
- Adds a `*.component.js` file that loads the templates and compiles them as a Dust component
- Updates file structure and naming to reflect the need to support multiple templates per component. This scheme treats the schema, ui, stylesheet and component loader as “main” files in the project root. Templates (and, potentially, subsidiary stylesheets) live in `lib/`.
- Adds tests that use Dust to render a spec file.
## How to test-drive this PR
- Clone this repo
- cd into `stencil-spec`
- run `npm install && bower install && bundle install`
- run `grunt serve` and take note of the port the server is on (defaults to 3000)
- open `http://localhost:{port}/tests/visual/`. If on port 3000: `http://localhost:3000/tests/visual/`
- verify that the tests render and pass (based on visual inspection)
## Research resources
- none
